src/pages/workspaces/workspace/Data.js

```diff
@@ -321,13 +324,19 @@ const DataTableActions = ({ workspace, tableName, rowCount, entityMetadata, onRe
           input({ type: 'hidden', name: 'FCtoken', value: getUser().token }),
           input({ type: 'hidden', name: 'model', value: 'flexible' })
         ]),
-        dataProvider.features.supportsTsvDownload && h(MenuButton, {
+        (dataProvider.features.supportsTsvDownload || dataProvider.features.supportsTsvAjaxDownload) && h(MenuButton, {
           disabled: isSetOfSets,
           tooltip: isSetOfSets ?
             'Downloading sets of sets as TSV is not supported at this time.' :
             'Download a TSV file containing all rows in this table.',
           onClick: () => {
-            downloadForm.current.submit()
+            if (dataProvider.features.supportsTsvDownload) {
+              downloadForm.current.submit()
+            } else if (dataProvider.features.supportsTsvAjaxDownload) {
+              // TODO: this overrides the filename specified by the WDS API. Is that ok?
+              dataProvider.downloadTsv(signal, tableName).then(blob => FileSaver.saveAs(blob, `${tableName}.tsv`))
+            }
```

Comment:
two comments on the TSV download:
1) we override filenames elsewhere in TSV download paths, so I'm assuming this is ok
2) we're using the `FileSaver` approach for WDS download because it prevents the need to support a cookie-authed or form-post-authed backend API. The `FileSaver` approach DOES work for Entity Service, but I am hesitant to make the change to something that is already working well.

Reply:
If we're going to use the Ajax download version for WDS, I'd vote for using it for both cases and scrapping the cookie authed endpoints.